사장님 로그인, 마이페이지 조회 & 수정, 로그아웃, 회원 탈퇴

shushuaka opened the pull request:
사장님 회원가입 기능 구현 #14
사장님 로그인 기능 구현#16
사장님 마이페이지 조회 #21
사장님 마이페이지 수정 #27
사장님 로그아웃 #37
사장님 회원 탈퇴 #39
OwnerService에 대한 단위 테스트 추가

JIMINJI1 committed:
feat(order-entity): orderStatus 기본값 설정 #5

JIMINJI1 committed:
feat(owner-entity): Enum OrderStatus 각 상태에 대한 설명 추가 #5

TIP-Santa committed:
See issue #9 feat(store-controller) : 가게 생성 기능 추가

TIP-Santa committed:
See issue #9 feat(store-create-req-dto) : store-create-req-dto 추가

TIP-Santa committed:
See issue #9 feat(store-service) : 가게 생성 시 가게이름 중복검사 추가

TIP-Santa committed:
See issue #9 feat(error-code) : 가게이름 중복검사 에러코드 추가

TIP-Santa committed:
See issue #9 feat(store-repository) : 가게이름 중복검사 추가

yaewon0411 committed:
feat(login-dto): 로그인 요청 dto 추가 #8

yaewon0411 committed:
feat(user-service): 유저 로그인 기능 추가 #8

yaewon0411 committed:
feat(user-controller): 유저 로그인 기능 추가 #8

yaewon0411 committed:
feat(valid-password): 어노테이션 제약 속성 추가 #8

yaewon0411 committed:
test(user-service): 로그인 단위 테스트 추가 #8
- 로그인 성공 케이스
- 로그인 실패 테스트: 존재하지 않는 이메일, 비밀번호 불일치

yaewon0411 committed:
test(user-controller): 로그인 단위 테스트 추가 #8

yaewon0411 committed:
refactor(dto-signup): 회원가입 시 전화번호 정규 표현식 검사 추가 #8

yaewon0411 committed:
refactor(user): 빌더로 생성 시 isDeleted 필드 false로 고정되도록 수정 #8

yaewon0411 committed:
feat(argument-resolver): 메서드 파라미터로 TokenSubject 반환하는 argument resolver 추가 #13

yaewon0411 committed:
feat(user-service): 마이페이지 조회 기능 추가 #13

yaewon0411 committed:
feat(user-controller): 마이페이지 조회 기능 추가 #13

yaewon0411 committed:
feat(config): argument resolver webMVc 설정 등록 #13

yaewon0411 committed:
fix(token-subject): 생성자 인식 문제 해결 위해 ConsturctorProperties 사용 #13

yaewon0411 committed:
test(user-service): 마이페이지 조회 단위 테스트 추가 #13

yaewon0411 committed:
test(user-controller): 마이페이지 조회 단위 테스트 추가 및 통합 테스트와 분리 #13

gbognon25 committed:
chore(ressources): "application.yaml" 및 "application-test.yaml" 수정 #10

yaewon0411 committed:
test(user-controller): 마이패이지 조회 단위 테스트 추가 #13

shushuaka committed:
feat(dto): OwnerJoinReqDto 생성 및 유효성 검사 추가 #14

- OwnerJoinReqDto 필드 정의
- 유효성 검사 어노테이션 추가

shushuaka committed:
feat(dto): OwnerJoinRespDto 생성 #14

- 회원가입 성공 시 성공 메시지를 반환

shushuaka committed:
feat(controller): 사장님 회원가입 API 추가 #14

- /api/owners/signup 추가
- 회원가입 요청 받아 OwnerService의 signup 메서드 호출
- 성공 시 OwnerJoinRespDto 반환

shushuaka committed:
feat(config): SecurityConfig 추가 및 PasswordEncoder 설정 #14

- PasswordEncoder 빈으로 BCryptPasswordEncoder 등록
- 비밀번호 암호화를 위한 Security 설정 파일 추가

gbognon25 committed:
feat(cart-entity): cart table 생성 #10

gbognon25 committed:
feat(category-entity): category table 생성 #10

gbognon25 committed:
feat(menu-entity): menu table 생성 #10

gbognon25 committed:
feat(order_menu-entity): order_menu table 생성 #10

gbognon25 committed:
feat(owner-entity): owner table 생성 #10

shushuaka committed:
feat(repo): 이메일 중복 검사 메서드 추가 #14

- 이메일 중복 확인을 위한 existsByEmail 메서드 추가

gbognon25 committed:
feat(review-entity): review table 생성 #10

gbognon25 committed:
feat(store-entity): store table 생성 #10

gbognon25 committed:
feat(store_category-entity): store_category table 생성 #10

shushuaka committed:
feat(service): OwnerService 회원가입 로직 및 비밀번호 암호화 추가 #14

- 회원가입 시 이메일 중복 검사 및 비밀버호 암호화
- 회원가입 성공 시 OwnerJoinRspDto 반환

shushuaka committed:
feat: Owner 엔티티 및 암호화 설정 추가 #14

- build.gradle에 Spring security 의존성 추가하여 비밀번호 암호화 지원
- Owner 엔티티 필드에 빌더 패턴 추가로 객체 생성 간편화

yaewon0411 committed:
feat(user-service): 유저 주소지 등록 기능 추가 #15

yaewon0411 committed:
feat(user-controller): 유저 주소지 등록 기능 추가 #15

gbognon25 committed:
feat(owner-entity): Enum OrderStatus 각 상태에 대한 설명 추가 #10

shushuaka committed:
refactor: OwnerService 수정 및 SecurityConfig 파일 삭제 #14

- 커스텀 PasswordEncoder 사용으로 OwnerService에서 Spring Security의 PasswordEncoder 의존성 제거
- 공통 PasswordEncoder로 변경하여 SecurityConfig 파일 삭제

yaewon0411 committed:
test(user-service): 주소지 등록 단위 테스트 추가 #15

yaewon0411 committed:
chore(gitignore): yaml gitignore #15

yaewon0411 committed:
test(user-controller): 주소지 등록 단위 테스트 추가 #15

shushuaka committed:
feat(dto): OwnerLoginReqDto 생성 #16

- 사장님 로그인 요청 데이터용 DTO 생성
- 이메일과 비밀번호 필드 추가

shushuaka committed:
feat(dto): OwnerLoginRespDto 생성 #16

- 로그인 성공 시 반환할 데이터용 DTO 생성
- 성공 메시지와 JWT 토큰 필드 추가

shushuaka committed:
feat(service): OwnerService에 로그인 메서드 추가 #16

- 로그인 로직을 처리하는 login 메서드 추가
- 이메일과 비밀번호 검증 후 JWT 토큰 반환

yaewon0411 committed:
feat(user-service): 주소지 수정 기능 추가 #19

yaewon0411 committed:
feat(user-controller): 주소지 수정 기능 추가 #19

shushuaka committed:
feat(repo): OwnerRepository에 findByEmail 메서드 추가 #16

- Optional<Owner> 타입으로 이메일 조회 기능 추가

shushuaka committed:
feat(controller): OwnerController에 로그인 API 추가 #16

- 사장님 로그인 요청을 처리할 POST /api/owners/login API 추가
- 로그인 요청 처리 후 OwnerLoginRespDto 반환

yaewon0411 committed:
feat(user-service): 주소지 수정 시 주소지 작성자 검증 로직 추가 #15

shushuaka committed:
feat(dto): 마이페이지 조회 응답 DTO 생성 #21

- 사장님 마이페이지 조회 시 반환할 OwnerMyPageRespDto 추가
- ID, 이름, 연락처, 이메일 정보를 포함

shushuaka committed:
feat(service): OwnerService에 마이페이지 조회 메서드 추가 #21

- 사장님의 ID를 통해 마이페이지 정보를 조회하는 getMyPage 메서드 추가
- 조회된 정보를 OwnerMyPageRespDto로 반환
- 사장님 정보가 없을 경우 예외 처리

shushuaka committed:
feat(controller): OwnerController에 마이페이지 조회 API 추가 #21

- 사장님 마이페이지 조회 요청을 처리할 GET /api/owners/mypage API 추가
- ownerId로 사장님 정보를 조회하여 OwnerMyPageRespDto로 응답

yaewon0411 committed:
test(user-service): 주소지 수정 단위 테스트 추가 #19

yaewon0411 committed:
refactor(address): 주소 수정 검증 로직 분리 #19

yaewon0411 committed:
feat(user-controller): 주소지 삭제 기능 추가 #23

yaewon0411 committed:
feat(user-service): 주소지 삭제 기능 추가 #23

yaewon0411 committed:
test(user-service): 주소지 삭제 단위 테스트 추가 #23

shushuaka committed:
refactor: 사장님 회원가입 기능 구현 리팩토링 및 예외 처리 추가 #14

- OwnerController 응답을 ApiResult로 통일하여 공통 포맷 적용
- OwnerService의 예외 처리에 BusinessException과 ErrorCode 사용
- OwnerJoinRespDto와 OwnerLoginRespDto에 기본 메시지 필드 추가
- OwnerLoginReqDto에 유효성 검증 메시지 추가 (이메일 및 비밀번호 형식)
- BusinessException, ServerException, CustomJwtException에 로깅 기능 추가

gbognon25 committed:
feat(cart-dto): cart-create-req-dto 추가

gbognon25 committed:
feat(cart-dto): cart-create-resp-dto 추가

gbognon25 committed:
feat(cart-dto): cart-create-req-dto 수정

gbognon25 committed:
feat(cart-dto): cart-create-resp-dto 수정

gbognon25 committed:
feat(cart-repo): cart-repository 추가

gbognon25 committed:
feat(cart-dto): cart-create-resp-dto 수정

gbognon25 committed:
feat(cart-dto): cart-create-req-dto 수정

gbognon25 committed:
feat(cart-service): cart-service 추가

gbognon25 committed:
feat(cart-repository): cart-repository 수정

gbognon25 committed:
feat(cart-repository): cart-repository 수정

gbognon25 committed:
feat(cart-service): cart-service 수정

gbognon25 committed:
feat(cart-controller): cart-controller 추가

yaewon0411 committed:
feat(user-controller): 전체 주소지 조회 기능 추가 #25

yaewon0411 committed:
feat(user-service): 전체 주소지 조회 기능 추가 #25

yaewon0411 committed:
test(user-service): 전체 주소지 조회 단위 테스트 추가 #25

gbognon25 committed:
feat(cart-entity): cart entity annotations 추가

gbognon25 committed:
feat(menu-entity): menu entity annotations 추가

gbognon25 committed:
feat(store-entity): store entity annotations 추가

yaewon0411 committed:
test(user-service): 전체 주소지 조회 단위 테스트 추가 #25

gbognon25 committed:
feat(cart-dto): cart-create-resp-dto에 method 추가

gbognon25 committed:
feat(cart-service): 장바구니 페이지 조회 단위 테스트 #28

yaewon0411 committed:
test(user-controller): 전체 주소지 조회 테스트 추가 #25

shushuaka committed:
refactor: OwnerService 및 ErrorCode 리팩토링 #14

- ErrorCode에 EMAIL_NOT_FOUND 및 OWNER_NOT_FOUND 추가하여 명확한 예외 처리
- OwnerService에서 EMAIL_NOT_FOUND와 OWNER_NOT_FOUND를 사용해 예외 메시지 개선
- @RequiredArgsConstructor로 OwnerService 생성자 간소화

yaewon0411 committed:
feat(user-service): 주소지 생성일 기준 내림차순 반환하도록 변경 #25

shushuaka committed:
Merge remote-tracking branch 'origin/dev' into hyejin

# Conflicts:
#	src/main/java/com/my/foody/global/ex/ErrorCode.java

yaewon0411 committed:
test(user-service): 주소지 조회 메서드명 변경 #25

shushuaka committed:
Merge remote-tracking branch 'origin/feat/cart/read' into hyejin

shushuaka committed:
Merge remote-tracking branch 'origin/feat/address/get-all' into hyejin

# Conflicts:
#	src/main/java/com/my/foody/global/ex/ErrorCode.java

shushuaka committed:
refactor(dto): OwnerJoinReqDto 유효성 검사 메시지 추가 #14

- 이름, 연락처, 이메일, 비밀번호 필드에 유효성 검사 메시지 추가
- 비밀번호는 최소 8자 이상, 최대 60자로 제한
- isDeleted 필드에 필수 입력 메시지 추가

shushuaka committed:
feat(dto): 사장님 마이페이지 수정 요청 및 응답 DTO 추가 #27

- OwnerMyPageUpdateReqDto: 마이페이지 수정 요청을 위한 DTO 생성
  - 이메일, 이름, 연락처, 현재 비밀번호, 새 비밀번호 필드 추가 및 유효성 검사 메시지 설정
- OwnerMyPageUpdateRespDto: 마이페이지 수정 응답을 위한 DTO 생성
  - id, 이름, 연락처, 이메일 필드 포함

shushuaka committed:
feat(service): 사장님 마이페이지 수정 기능 추가 및 엔티티 업데이트 메서드 구현 #27

- OwnerService에 사장님 마이페이지 수정 기능 추가
  - 현재 비밀번호 확인 후 새 정보로 업데이트
  - 업데이트된 정보를 OwnerMyPageUpdateRespDto로 반환
  - @Transactional로 데이터의 일관성 유지
- Owner 엔티티에 updateInfo 메서드 추가
  - 이름, 연락처, 이메일, 비밀번호를 새로운 값으로 업데이트 가능

shushuaka committed:
feat(controller): 사장님 마이페이지 수정 API 추가 및 기존 기능 개선 #27

- 사장님 마이페이지 수정 API (/mypage) 추가
  - @PatchMapping으로 기존 정보 수정 가능
  - 수정된 정보는 OwnerMyPageUpdateRespDto로 반환

shushuaka committed:
refactor: ErrorCode에 OWNER_NOT_FOUND 추가 #27

- 사장님 조회 시 존재하지 않는 경우를 처리하기 위해 OWNER_NOT_FOUND 오류 코드 추가

shushuaka committed:
Merge remote-tracking branch 'origin/dev' into hyejin

# Conflicts:
#	src/main/java/com/my/foody/domain/user/controller/UserController.java
#	src/main/java/com/my/foody/domain/user/service/UserService.java
#	src/test/java/com/my/foody/domain/user/controller/UserControllerTest.java
#	src/test/java/com/my/foody/domain/user/service/UserServiceTest.java

shushuaka committed:
feat(dto): 사장님 로그아웃 응답 DTO 추가 #37

- OwnerLogoutRespDto 생성
- 로그아웃 성공 시 "로그아웃이 완료되었습니다." 메시지 반환

shushuaka committed:
feat(controller): 사장님 로그아웃 API 추가 #37

- POST /api/owners/logout API 추가
- 로그아웃 성공 시 OwnerLogoutRespDto 반환

shushuaka committed:
feat(service): 사장님 로그아웃 메서드 추가 #37

- OwnerService에 로그아웃 메서드 추가
- 로그아웃 성공 시 OwnerLogoutRespDto 반환

shushuaka committed:
refactor: ErrorCode enum 불필요한 세미콜론 제거 및 코드 정리

shushuaka committed:
feat: 사장님 회원 탈퇴 요청 DTO 추가 #39

- OwnerDeleteReqDto 생성하여 회원 탈퇴 시 비밀번호 검증 기능 추가
- 비밀번호 형식 검증: 최소 8자, 60자 이하, 하나 이상의 문자와 숫자 포함

shushuaka committed:
feat: 사장님 회원 탈퇴 응답 DTO 추가 #39

- OwnerDeleteRespDto 추가: 회원 탈퇴 시 고정된 메시지를 제공하는 싱글톤 인스턴스 적용

shushuaka committed:
feat(OwnerService, Owner): 사장님 회원 탈퇴 기능 추가 #39

- OwnerService에 deleteOwner 메서드 구현
- Owner 엔티티에 markAsDeleted 메서드 추가하여 삭제 상태 설정

shushuaka committed:
refactor: 비밀번호 유효성 검증 로직 통일화

- OwnerJoinReqDto, OwnerLoginReqDto, OwnerMyPageUpdateReqDto의 비밀번호 유효성 검증을 통일하여 일관성 유지

shushuaka committed:
merge: dev에서 hyejin 브랜치로 병합

shushuaka committed:
refactor: StoreServiceTest 코드 리팩토링 (불필요한 import 제거)

shushuaka committed:
Merge remote-tracking branch 'origin/dev' into hyejin

# Conflicts:
#	src/main/java/com/my/foody/domain/owner/entity/Owner.java
#	src/main/java/com/my/foody/domain/user/dto/req/valid/ValidPassword.java
#	src/main/java/com/my/foody/global/ex/ErrorCode.java
#	src/main/java/com/my/foody/global/util/DummyObject.java
#	src/test/java/com/my/foody/domain/user/controller/UserControllerTest.java
#	src/test/java/com/my/foody/domain/user/service/UserServiceTest.java

shushuaka committed:
feat(controller): 사장님 회원 탈퇴 API 추가 #39

- DELETE /api/owners 엔드포인트 추가
- OwnerDeleteReqDto를 통한 비밀번호 검증 후 탈퇴 처리
- 탈퇴 성공 시 OwnerDeleteRespDto 반환

shushuaka committed:
fix(bug): 병합 충돌 해결

shushuaka committed:
fix(bug): OrderStatus enum 생성자 추가로 초기화 오류 해결

shushuaka committed:
fix: Owner 엔티티에 정보 업데이트 및 삭제 상태 메서드 추가로 오류 해결

shushuaka committed:
chore: Spring Security Test 의존성 build.gradle에 추가

shushuaka committed:
chore: Owner 요청 DTO에 @AllArgsConstructor 추가

- (OwnerDeleteReqDto, OwnerJoinReqDto, OwnerLoginReqDto, OwnerMyPageUpdateReqDto)

shushuaka committed:
fix: 사장님 탈퇴 처리 시 삭제 상태를 데이터베이스에 반영하도록 수정 #39

shushuaka committed:
test: OwnerService에 대한 단위 테스트 추가 #16, #21, #27, #37, #39

shushuaka committed:
Merge remote-tracking branch 'origin/dev' into hyejin

# Conflicts:
#	src/test/java/com/my/foody/domain/store/service/StoreServiceTest.java

shushuaka committed:
Merge pull request #50 from (dev)A2jo/feat/owner/get-all-store

yaewon0411 committed:
feat(cart-menu): CartMenu 엔티티 생성 #60

yaewon0411 committed:
feat(cart-service): 장바구니 메뉴 담기 구현 중 #60

yaewon0411 committed:
test(cart-service): 장바구니 아이템 담기 단위 테스트 추가 #60

yaewon0411 committed:
feat(menu-service): 주문 가능한 메뉴 검증 로직 보강 #60

yaewon0411 committed:
test(cart-service) 장바구니 메뉴 담기 단위 테스트 추가 #60

yaewon0411 committed:
test(cart-controller): 장바구니 메뉴 담기 단위 테스트 추가 #60

shushuaka committed:
Merge remote-tracking branch 'origin/dev' into hyejin

# Conflicts:
#	src/main/java/com/my/foody/global/ex/ErrorCode.java
#	src/test/java/com/my/foody/domain/order/service/OrderServiceTest.java

shushuaka committed:
feat: 메인 화면 조회 API 추가 #67

- 메인 화면에서 카테고리와 가게 목록을 반환하는 API 추가

shushuaka committed:
feat: UserHomeRespDto 클래스 추가해서 메인 화면 응답 DTO 구성 #67

- 카테고리 및 가게 정보를 포함하는 UserHomeRespDto 생성
- CategoryDto, StoreDto 내부 클래스로 카테고리와 가게 정보를 각각 DTO로 변환

shushuaka committed:
feat: UserService에 메인 화면 조회 기능 추가 #67

shushuaka committed:
Merge remote-tracking branch 'origin/dev' into hyejin

# Conflicts:
#	src/main/java/com/my/foody/global/ex/ErrorCode.java

shushuaka committed:
Merge remote-tracking branch 'origin/hyejin' into hyejin

# Conflicts:
#	src/main/java/com/my/foody/global/ex/ErrorCode.java

shushuaka committed:
Merge remote-tracking branch 'origin/dev' into hyejin

# Conflicts:
#	src/main/java/com/my/foody/domain/owner/repo/OwnerRepository.java
#	src/main/java/com/my/foody/domain/owner/service/OwnerService.java
#	src/main/java/com/my/foody/domain/user/service/UserService.java
#	src/main/java/com/my/foody/global/ex/ErrorCode.java
#	src/test/java/com/my/foody/domain/store/service/StoreServiceTest.java

shushuaka committed:
fix(merge): 병합 충돌 수정

shushuaka committed:
fix: MenuService에서 @Transactional 어노테이션 오타 수정

shushuaka committed:
Merge remote-tracking branch 'origin/dev' into hyejin

# Conflicts:
#	src/main/java/com/my/foody/domain/menu/service/MenuService.java
#	src/main/java/com/my/foody/global/ex/ErrorCode.java

JIMINJI1 reviewed (approved):
확인했습니다.

shushuaka committed:
refactor: 메인화면 조회 도메인 home으로 이동을 위해 기존 코드 및 파일 삭제 #67

yaewon0411 commented on src/main/java/com/my/foody/domain/owner/controller/OwnerController.java:
```diff
@@ -1,4 +1,68 @@
+    }
+
+    @GetMapping("/mypage")
+    public ResponseEntity<ApiResult<OwnerMyPageRespDto>> getMyPage(@AuthenticationPrincipal Long ownerId) {
```
이건 시큐리티를 사용할 때 쓰는 어노테이션입니다
저희는 @CurrentUser를 통해 TokenSubject로 인터셉터에서 설정한 사용자 정보 받아오고 있습니다

yaewon0411 commented on src/main/java/com/my/foody/domain/owner/controller/OwnerController.java:
```diff
@@ -1,4 +1,68 @@
+            @RequestBody OwnerDeleteReqDto deleteReqDto) {
+
+        OwnerDeleteRespDto responseMessage = ownerService.deleteOwner(ownerId, deleteReqDto);
+        return ResponseEntity.ok(ApiResult.success(responseMessage));
```
RequestBody는  @Valid와 같이 사용하지 않으면 유효성 검사가 진행되지 않습니다

yaewon0411 commented on src/main/java/com/my/foody/domain/owner/controller/OwnerController.java:
```diff
@@ -1,4 +1,68 @@
+        OwnerLoginRespDto responseMessage = ownerService.login(ownerLoginReqDto);
+        return ResponseEntity.ok(ApiResult.success(responseMessage));
+    }
+
```
토큰은 응답 바디가 아니라 응답 헤더에 나가야 합니다
수정 바랍니다

yaewon0411 commented on src/main/java/com/my/foody/domain/owner/dto/req/OwnerMyPageUpdateReqDto.java:
```diff
@@ -0,0 +1,33 @@
+@AllArgsConstructor
+public class OwnerMyPageUpdateReqDto {
+
+    @Email(message = "올바른 이메일 형식이 아닙니다.")
+    @NotBlank(message = "이메일은 필수 입력 항목입니다.")
+    private String email;
+
+    @NotBlank(message = "이름은 필수 입력 항목입니다.")
+    @Size(max = 100, message = "이름은 최대 100자까지 가능합니다.")
+    private String name;
+
+    @Size(max = 15, message = "전화번호는 최대 15자까지 가능합니다.")
+    private String contact;
+
+    @NotBlank(message = "현재 비밀번호는 필수 입력 항목입니다.")
+    private String currentPassword;
+
+    @NotBlank(message = "비밀번호는 필수 입력 항목입니다.")
+    @Size(min = 8, max = 60, message = "새 비밀번호는 8자 이상, 60자 이하로 설정해야 합니다.")
+    @Pattern(regexp = "^(?=.*[A-Za-z])(?=.*\\d)[A-Za-z\\d]{8,}$",
+            message = "새 비밀번호는 최소 8자, 하나 이상의 문자와 숫자를 포함해야 합니다.")
+    private String newPassword;
```
모든 필드를 다 NotBlank로 받고 계신데 이전에 와이어프레임에서 논의했던 시나리오와 다릅니다

현재는 FE에서 클라이언트가 수정한 내용과 수정하지 않은 내용 모두 보내는 것으로 시나리오를 설정하신 것으로 보입니다

후자를 의도하신 게 맞다면 말씀 주세요

shushuaka replied:
아 업데이트인데... 그렇네요 사용자가 수정하고 싶은 내용만 클라이언트에서 선택해서 보낼 수 있도록 @NotBlank를 제거하겠습니다.

yaewon0411 commented on src/main/java/com/my/foody/domain/owner/service/OwnerService.java:
```diff
@@ -1,25 +1,133 @@
+    // 로그아웃 메서드 추가
+    public OwnerLogoutRespDto logout() {
+        return new OwnerLogoutRespDto();  // 로그아웃 성공 메시지 반환
+    }
```
로그아웃 기능이 누락되었습니다

yaewon0411 commented on src/main/java/com/my/foody/domain/owner/service/OwnerService.java:
```diff
@@ -1,25 +1,133 @@
+    // 로그인 메서드 추가
+    public OwnerLoginRespDto login(OwnerLoginReqDto reqDto) {
+        // 1. 이메일로 Owner 조회
+        Owner owner = ownerRepository.findByEmail(reqDto.getEmail())
+                .orElseThrow(() -> new BusinessException(ErrorCode.EMAIL_NOT_FOUND));
```
해당 owner가 이미 isDeleted된 회원인지 아닌지를 검증해야 합니다

yaewon0411 reviewed (commented):
👍
테스트 케이스가 통과되는 것 확인했습니다
현재는 컨트롤러 쪽 테스트가 필요해 보입니다

우선 @Valid와 @CurrentUser를 통한 유효성 검사/권한 검사를 적용한 후 컨트롤러 단위 테스트나 포스트맨으로의 테스트 부탁드립니다

shushuaka committed:
feat: 메인 화면 조회용 DTO 생성 #67

shushuaka committed:
refactor: Category 엔티티에 생성자 및 toString 추가 #67

shushuaka committed:
feat: 메인 화면 카테고리 목록 조회 기능 구현 #67

shushuaka committed:
feat: HomeController 메인 화면 조회 API 추가 #67

shushuaka committed:
test: HomeController 메인 화면 조회 API 테스트 코드 추가 #67

shushuaka committed:
Merge remote-tracking branch 'origin/dev' into hyejin

# Conflicts:
#	src/main/java/com/my/foody/global/ex/ErrorCode.java

shushuaka committed:
feat: 마이페이지 조회 시 @CurrentUser를 통해 TokenSubject로 사용자 정보 받아오도록 수정 #21

- @AuthenticationPrincipal 대신 @CurrentUser를 사용하여 TokenSubject에서 ownerId를 가져오도록 변경

shushuaka committed:
feat: OwnerController에 @Valid 추가 및 @CurrentUser로 변경

- signup, login, updateMyPage, deleteOwner 메서드에 @Valid 추가하여 유효성 검사가 적용되도록 설정

shushuaka committed:
fix: Owner 로그인 시 JWT 토큰을 응답 헤더에 추가하도록 수정

- 로그인 성공 시 Authorization 헤더에 JWT 토큰 포함
- 응답 본문에는 기존 로그인 성공 메시지 유지

shushuaka committed:
refactor: OwnerMyPageUpdateReqDto에서 모든 필드 필수 조건 제거

- 이메일, 이름, 연락처, 비밀번호 필드의 @NotBlank 어노테이션 제거
- 클라이언트가 수정할 항목만 선택적으로 보낼 수 있도록 변경

shushuaka committed:
feat: 로그아웃 시 토큰 무효화 및 블랙리스트 체크 기능 추가

- JwtProvider: 로그아웃 시 토큰을 블랙리스트에 추가하는 `invalidateToken` 메서드 추가
- OwnerService: 로그아웃 메서드에서 JWT 무효화를 위해 `invalidateToken` 호출
- OwnerController: 로그아웃 엔드포인트에서 응답 헤더에 토큰 제거

shushuaka committed:
refactor: 불필요한 중복된 오류 코드 제거 및 코드 정리

- ErrorCode에서 중복된 OWNER_NOT_FOUND 코드 제거
